dev/tasks/linux-packages/apache-arrow/yum/centos-8-stream/Dockerfile

```
@@ -18,6 +18,10 @@
 ARG FROM=quay.io/centos/centos:stream8
 FROM ${FROM}
 
+ENV \
+  GCC_MAJOR=11 \
+  SCL=gcc-toolset-11
```

Can we try GCC 12 (or 13) on CentOS Stream 8?

```suggestion
  GCC_MAJOR=12 \
  SCL=gcc-toolset-12
```

---

I then get the same error when linking the Vala stuff:
```
[107/130] Linking target example/vala/build
FAILED: example/vala/build 
c++  -o example/vala/build example/vala/build.p/meson-generated_build.c.o -Wl,--as-needed -Wl,--no-undefined -Wl,-z,relro -Wl,-z,now -specs=/usr/lib/rpm/redhat/redhat-hardened-ld -O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fexceptions -fstack-protector-strong -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection '-Wl,-rpath,$ORIGIN/../../arrow-glib:$ORIGIN/../../../../cpp/build/release' -Wl,-rpath-link,/build/rpmbuild/BUILD/apache-arrow-20.0.0.dev56/c_glib/build/arrow-glib -Wl,-rpath-link,/build/rpmbuild/BUILD/apache-arrow-20.0.0.dev56/c_glib/build/../../cpp/build/release -Wl,--start-group arrow-glib/libarrow-glib.so.2000.0.0 /usr/lib64/libgio-2.0.so /usr/lib64/libgobject-2.0.so /usr/lib64/libglib-2.0.so -Wl,--end-group
/opt/rh/gcc-toolset-12/root/usr/libexec/gcc/x86_64-redhat-linux/12/ld: /opt/rh/gcc-toolset-12/root/usr/lib/gcc/x86_64-redhat-linux/12/libstdc++_nonshared.a(functexcept80.o): undefined reference to symbol '_ZTISt20bad_array_new_length@@CXXABI_1.3.8'
/opt/rh/gcc-toolset-12/root/usr/libexec/gcc/x86_64-redhat-linux/12/ld: /usr/lib64/libstdc++.so.6: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
```

Can I ask you to take a closer look? The Vala build chain is totally outside of my competence (I didn't even know we have a Vala build somewhere).

---

OK. I'll take over this part.